examples/example.json

```diff
@@ -0,0 +1,234 @@
+                    },
+                    {
+                        "value": 7,
+                        "name": "Residential Buildings",
+                        "description": null,
+                        "title": null,
+                        "color_hint": null,
+                        "nodata": false
+                    },
+                    {
+                        "value": 8,
+                        "name": "River",
+                        "description": null,
+                        "title": null,
+                        "color_hint": null,
+                        "nodata": false
+                    },
+                    {
+                        "value": 9,
+                        "name": "SeaLake",
+                        "description": null,
+                        "title": null,
+                        "color_hint": null,
+                        "nodata": false
+                    }
+                ],
+                "post_processing_function": null
+            }
+        ],
+        "mlm:runtime": [
+            {
+                "asset": {
+                    "href": "https://huggingface.co/torchgeo/resnet18_sentinel2_all_moco/resolve/main/resnet18_sentinel2_all_moco-59bfdff9.pth",
+                    "title": null,
+                    "description": null,
+                    "type": null,
+                    "roles": null
+                },
+                "source_code": {
+                    "href": "https://github.com/microsoft/torchgeo/blob/61efd2e2c4df7ebe3bd03002ebbaeaa3cfe9885a/torchgeo/models/resnet.py#L207",
+                    "title": null,
+                    "description": null,
+                    "type": null,
+                    "roles": null
+                },
+                "accelerator": "cuda",
+                "accelerator_constrained": false,
+                "hardware_summary": "Unknown",
+                "container": null,
+                "commit_hash": "61efd2e2c4df7ebe3bd03002ebbaeaa3cfe9885a",
+                "batch_size_suggestion": null
+            }
+        ],
+        "mlm:total_parameters": 11700000,
+        "mlm:pretrained_source": "EuroSat Sentinel-2",
+        "mlm:summary": "Sourced from torchgeo python library,identifier is ResNet18_Weights.SENTINEL2_ALL_MOCO"
+    },
+    "geometry": "{\n  \"type\": \"Polygon\",\n  \"coordinates\": [\n    [\n      [\n        -7.882190080512502,\n        37.13739173208318\n      ],\n      [\n        -7.882190080512502,\n        58.21798141355221\n      ],\n      [\n        27.911651652899923,\n        58.21798141355221\n      ],\n      [\n        27.911651652899923,\n        37.13739173208318\n      ],\n      [\n        -7.882190080512502,\n        37.13739173208318\n      ]\n    ]\n  ]\n}",
+    "links": [
+        {
+            "rel": "derived_from",
+            "href": "https://earth-search.aws.element84.com/v1/collections/sentinel-2-l2a",
+            "type": "application/json"
+        }
```

Comment:
Consider adding `ml-aoi:split: train` to make an explicit reference that this corresponds to the training samples used for generating the model (see https://cloudnativegeo.slack.com/archives/C06559H0UUQ/p1710265427364609?thread_ts=1710251736.833979&cid=C06559H0UUQ).

A `title` mentioning this could be relevant also to make the example link more obvious by inexperienced users with all the STAC extensions involved. 

I think this capability definitely merits an improved `add_derived_from` optional parameter to help users generate the item (`ml_aoi_split: ML_AOI_Split = None` ?). 
for the type, see: https://github.com/stac-extensions/ml-aoi/pull/8/files#diff-2c01497f2468c2d24ba1055b1df7c2e3fe570e38c3b0804dee337c199c259024R66

Reply:
I'm making a separate issue for this since this PR is getting long if that's ok. can be addressed post version 2 I think?